### PR TITLE
{Error Improvement} Fix #14830: TypeError related issues

### LINKED ISSUE #14830: [Error] TypeError from code
**Describe the bug**

This issue is acting as a track of all "TypeError" from github, since we're working on fixing TypeError.

**Examples**
- string indices must be integers

### PR DESCRIPTION
**Pull request by houk-ms**
**Description**  

This PR fixes TypeError related issues in CLI codes. TypeErrors are abvious code bugs which frustrate users a lot. An example coud be 

```
...
trace 
...
TypeError: string indices must be integers
```

It does not appear all the time but in some corner cases. It usually happens when users use CLI in a wrong way. But as a better way to handle and categorized the errors, we should check the corner cases and raise proper error types.

**Testing Guide**  

As the issues in #14830

### CODE REVIEW

**Review by yungezz — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/util.py [hunk 1]
```diff
@@ -89,8 +89,11 @@ def handle_exception(ex):  # pylint: disable=too-many-return-statements
             return 1
         if isinstance(ex, HttpOperationError):
             try:
-                response_dict = json.loads(ex.response.text)
-                error = response_dict['error']
+                response = json.loads(ex.response.text)
+                if isinstance(response, str):
+                    error = response
+                else:
+                    error = response['error']
```
Comment (bim-msft) [inline src/azure-cli-core/azure/cli/core/util.py:96]: What if the response type is a list?

Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/util.py:96]: I believe it's impossible, it's already weird enough being a str

**Review by bim-msft — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/util.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli-core/azure/cli/core/util.py:117]: Is it possible 'error' is not in response?

Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/util.py:117]: It's decided by service response, so I am not sure. But currently, we only found the cases that response is a tring type.

**Review by qwordy — APPROVED** [review]

**Review by arrownj — APPROVED** [review]
👍

**Comment by houk-ms** [comment]
#15314

**Pull request merged**